lib/tlscontext.h

```diff
@@ -44,6 +44,16 @@ typedef enum
   TVM_REQUIRED=0x0020,
 } TLSVerifyMode;
 
+typedef enum
+{
+  TSO_NONE,
+  TSO_NOSSLv2=0x0001,
+  TSO_NOSSLv3=0x0002,
+  TSO_NOTLSv1=0x0004,
+  TSO_NOTLSv11=0x0008,
+  TSO_NOTLSv12=0x0010,
+} TLSSslOptions;
+
```

Comment:
a number of this calls were converted to enums recently. I did start to
move to the enum direction. An example is TLSVerifyMode just above this
declaration. Or msg-format.h, or a number of others. So this is not
consistent. I do know that I was converting a couple of these.

In these cases the call sites do not use the enum type (I don't even see
the use for the typedef itself, that should be abandoned), the the
call-sites should use "int"


-- 
Bazsi

On Wed, May 20, 2015 at 8:09 PM, juhaszviktor <notifications@github.com>
wrote:

> In lib/tlscontext.h
> <https://github.com/balabit/syslog-ng/pull/457#discussion_r30731421>:
>
> > @@ -44,6 +44,16 @@ typedef enum
> >    TVM_REQUIRED=0x0020,
> >  } TLSVerifyMode;
> >
> > +typedef enum
> > +{
> > +  TSO_NONE,
> > +  TSO_NOSSLv2=0x0001,
> > +  TSO_NOSSLv3=0x0002,
> > +  TSO_NOTLSv1=0x0004,
> > +  TSO_NOTLSv11=0x0008,
> > +  TSO_NOTLSv12=0x0010,
> > +} TLSSslOptions;
> > +
>
> I can agree your standpoint, but if I look at the whole syslog-ng code,
> the flags and some kind of options are declared as a define. For example in
> logwriter.h LW_* and LWO_* are also defines.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/457/files#r30731421>.
>